modules/siddhi-core/src/main/java/org/wso2/siddhi/core/query/SelectInsertStoreQueryRuntime.java

```diff
@@ -0,0 +1,135 @@
+        this.setOutputAttributes(metaStreamEvent.getLastInputDefinition().getAttributeList());
+    }
+
+    @Override
+    public Event[] execute() {
+        try {
+            StateEvent stateEvent = new StateEvent(1, outputAttributes.length);
+            StreamEvent streamEvent = new StreamEvent(metaStreamEvent.getBeforeWindowData().size(),
+                    metaStreamEvent.getOnAfterWindowData().size(),
+                    metaStreamEvent.getOutputData().size());
+            stateEvent.addEvent(0, streamEvent);
+
+            ComplexEventChunk complexEventChunk = new ComplexEventChunk(stateEvent, stateEvent, true);
+
+            if (eventType == MetaStreamEvent.EventType.TABLE) {
+                selector.process(complexEventChunk);
+            }
+            return new Event[]{};
+        } catch (Throwable t) {
+            throw new StoreQueryRuntimeException("Error executing '" + queryName + "', " + t.getMessage(), t);
+        }
+    }
+
+    @Override
+    public void reset() {
+        if (selector != null) {
+            selector.process(generateResetComplexEventChunk(metaStreamEvent));
+        }
+    }
+
+    /**
+     * This method sets a state event pool for the select insert query runtime.
+     *
+     * @param stateEventPool stateEventPool for the store query runtime
+     */
+    public void setStateEventPool(StateEventPool stateEventPool) {
+        if (stateEventPool != null) {
+            this.stateEventPool = stateEventPool;
+        } else {
+            throw new StoreQueryRuntimeException("Error occured while initializing run time for update or insert into" +
+                    "  query \"" + queryName + "\".");
+        }
+    }
+
+    /**
+     * This method sets selector for the select insert store query runtime.
+     *
+     * @param selector for the store query
+     */
+    public void setSelector(QuerySelector selector) {
+        this.selector = selector;
+    }
+
+    /**
+     * This method sets the output attribute list of the given store query.
+     *
+     * @param outputAttributeList output attritbutes of the store query
+     */
+    public void setOutputAttributes(List<Attribute> outputAttributeList) {
+        this.outputAttributes = outputAttributeList.toArray(new Attribute[outputAttributeList.size()]);
+    }
+
+    @Override
+    public Attribute[] getStoreQueryOutputAttributes() {
+        return Arrays.copyOf(outputAttributes, outputAttributes.length);
+    }
+
+    private ComplexEventChunk<ComplexEvent> generateResetComplexEventChunk(MetaStreamEvent metaStreamEvent) {
+        StreamEvent streamEvent = new StreamEvent(metaStreamEvent.getBeforeWindowData().size(),
+                metaStreamEvent.getOnAfterWindowData().size(), metaStreamEvent.getOutputData().size());
+        streamEvent.setType(ComplexEvent.Type.RESET);
+
+        StateEvent stateEvent = stateEventPool.borrowEvent();
+        if (eventType == MetaStreamEvent.EventType.AGGREGATE) {
```

Comment:
I think this will not be used at all, remove this

Reply:
done with commit a9c25b9